src/AppInstallerCLICore/Commands/PinCommand.cpp

```diff
@@ -113,16 +114,28 @@ namespace AppInstaller::CLI
 
     void PinAddCommand::ExecuteInternal(Execution::Context& context) const
     {
+        if (context.Args.Contains(Execution::Args::Type::Id))
+        {
+            // When we are given an ID, just pin that available package without checking for installed.
+            // This helps when there are matching issues, for example due to multiple side-by-side installs.

```

Comment:
> So previously, our goal for pinning available package is: if we find one available package to pin, all other available packages from other sources are pinned, except if user provides -s <source>
>
> Do you think we should do this for --id pins as well?

I hadn't thought about that. With the code as it is, it will add pins to all sources matching the query (--id or otherwise). So if you do `pin add Microsoft.PowerToys` and multiple sources have `Microsoft.PowerToys` we'd be adding pins to all of them, which seems okay to me. But if the ID in one of the sources is just "PowerToys" we wouldn't pin it. That also seems okay to me because we can't guess what queries we'd have to make to match the package from another source as it can be totally different. And this applies whether we are using --id or some other filter

Reply:
We could construct a query for systemReferenceStrings to find available packages from other sources. We can do it later if such asks emerges. But the behavior is different from what I specc'ed before.